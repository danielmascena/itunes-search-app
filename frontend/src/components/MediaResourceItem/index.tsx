import * as React from 'react';
import ListItem from "@mui/material/ListItem";
import ListItemText from "@mui/material/ListItemText";
import ListItemAvatar from "@mui/material/ListItemAvatar";
import Avatar from "@mui/material/Avatar";
import Typography from "@mui/material/Typography";

import {getMediaDescription} from "../../helpers";
interface Props {
  media: Media;
}

const MediaResourceItem: React.FC<Props> = ({ media }) => {
    const {
        artistName,
        artworkUrl100
    } = media;
    const [primaryText, secondaryText, extraText] = getMediaDescription(media);

  return (
    <ListItem alignItems="flex-start">
      {artworkUrl100 && (
          <ListItemAvatar>
            <Avatar alt={artistName} src={artworkUrl100} />
        </ListItemAvatar>
      )}
      <ListItemText
        primary={primaryText}
        secondary={
          <React.Fragment>
            <Typography
              sx={{ display: "inline" }}
              component="span"
              variant="body2"
              color="text.primary"
            >
              {secondaryText}
            </Typography>
            {extraText}
          </React.Fragment>
        }
      />
    </ListItem>
  );
};

export default MediaResourceItem;
