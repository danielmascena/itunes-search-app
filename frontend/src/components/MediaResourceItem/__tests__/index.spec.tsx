import * as React from "react";
import { render, screen } from "@testing-library/react";
import { v4 as uuidv4 } from 'uuid';

import MediaResourceItem from "../index";

jest.mock("uuidv4");
describe("MediaResourceItem Component", () => {

    it.todo("should render the markup");

    it("renders the song name", () => {
        const mockMedia = {
            artistName: "Iron Maiden",
            artworkUrl100: "",
            mediaType: "artist",
            artistId: uuidv4(),
            primaryGenreName: "metal"
        };

        render(<MediaResourceItem {...mockMedia} />);
        const linkElement = screen.getByText(/iron maiden/i);
        expect(linkElement).toBeInTheDocument();
    });
});