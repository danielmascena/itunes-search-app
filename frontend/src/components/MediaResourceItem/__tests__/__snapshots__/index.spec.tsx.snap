// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaResourceItem Component renders the MediaResourceItem component 1`] = `
<li
  className="MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-alignItemsFlexStart css-hhvo2v-MuiListItem-root"
  disabled={false}
>
  <div
    className="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
  >
    <span
      className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
    >
      Iron Maiden - artist
    </span>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
    >
      <span
        className="MuiTypography-root MuiTypography-body2 css-1phucj-MuiTypography-root"
      >
        artist/iron-maiden/546381
      </span>
       Artist genre: metal
    </p>
  </div>
</li>
`;
